model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py

```diff
@@ -0,0 +1,646 @@
+              configurable activations.
+            w_qcs: weights quantization candidates for custom bit-width mode. Must provide configuration for all
+              configurable weights.
+            metrics: metrics to include for computation. If None, all metrics are calculated.
+
+        Returns:
+            Resource utilization object.
+        """
+        metrics = set(metrics) if metrics else set(RUTarget)
+
+        w_total, a_total = None, None
+        if {RUTarget.WEIGHTS, RUTarget.TOTAL}.intersection(metrics):
+            w_total, *_ = self.compute_weights_utilization(target_criterion, bitwidth_mode, w_qcs)
+        elif w_qcs is not None:    # pragma: no cover
+            raise ValueError('Weight configuration passed but no relevant metric requested.')
+
+        if act_qcs and not {RUTarget.ACTIVATION, RUTarget.TOTAL}.intersection(metrics):    # pragma: no cover
+            raise ValueError('Activation configuration passed but no relevant metric requested.')
+        if RUTarget.ACTIVATION in metrics:
+            a_total, *_ = self.compute_activations_utilization(target_criterion, bitwidth_mode, act_qcs)
+
+        ru = ResourceUtilization()
+        if RUTarget.WEIGHTS in metrics:
+            ru.weights_memory = w_total
+        if RUTarget.ACTIVATION in metrics:
+            ru.activation_memory = a_total
+        if RUTarget.TOTAL in metrics:
+            # TODO use maxcut
+            act_tensors_total, *_ = self.compute_activation_tensors_utilization(target_criterion, bitwidth_mode, act_qcs)
+            ru.total_memory = w_total + act_tensors_total
+        if RUTarget.BOPS in metrics:
+            ru.bops, _ = self.compute_bops(target_criterion=target_criterion,
+                                           bitwidth_mode=bitwidth_mode, act_qcs=act_qcs, w_qcs=w_qcs)
+
+        assert ru.get_restricted_metrics() == set(metrics), 'Mismatch between the number of requested and computed metrics'
+        return ru
+
+    def compute_weights_utilization(self,
+                                    target_criterion: TargetInclusionCriterion,
+                                    bitwidth_mode: BitwidthMode,
+                                    w_qcs: Optional[Dict[BaseNode, NodeWeightsQuantizationConfig]] = None) \
+            -> Tuple[float, Dict[BaseNode, Utilization], Dict[BaseNode, Dict[str, Utilization]]]:
+        """
+        Compute graph's weights resource utilization.
+
+        Args:
+            target_criterion: criterion to include targets for computation.
+            bitwidth_mode: bit-width mode for computation.
+            w_qcs: weights quantization config per node for the custom bit mode. Must provide configuration for all
+              configurable weights.
+
+        Returns:
+            - Total weights utilization.
+            - Per node total utilization. Dict keys are nodes in a topological order.
+            - Detailed per node per weight utilization. Dict keys are nodes in a topological order.
```

Comment:
per weight "attribute". and in the one above also mention "sum of all weight attributes utilization" or something like that